doc/release-process.md

```diff
@@ -0,0 +1,40 @@
+# Weave Gitops Release Process
+
+The current release process for weave gitops is fairly straightforward. You need to:
+- Create the actual release
```

Comment:
I think we can also push a tag to trigger this. But I could be confused. At least that is how I was doing the releases last time I modify `go-releaser` 🙂

Reply:
Quite possibly; this seems a bit more obvious to me. :man_shrugging: